shoes-core/lib/shoes/http_wrapper.rb

```diff
@@ -0,0 +1,55 @@
+require 'uri'
+require 'net/http'
+
+# Wrapper for Net::HTTP to provide us to the functionality that we need which
+# isn't easily available from any particular gem out there :(
+#
+# Specific things this class does that we need include:
+#
+#   1) Reading in chunks to trigger progress
+#   2) Allow for graceful redirection
+#
+# Most of the gems out there do 2) but not 1), or do both but are untested on
+# JRuby.
+#
+class Shoes
+  class HttpWrapper
+    def execute(url, meth, body, headers = {}, redirects_left = 5, &blk)
+      uri = URI(url)
```

Comment:
It's probably fine... I know Avdi likes them `Array(...)` methods etc. and that's fine and I think we should use them more, as URI looks more like a constant (but it's probably a class/module) it looked odd/firghtening/why-ish to me but I'm good with it :)